core/src/main/java/org/apache/carbondata/core/util/DataTypeUtil.java

```diff
@@ -775,4 +775,25 @@ public static DataType valueOf(String name) {
     }
   }
 
-}
\ No newline at end of file
+  /**
+   * Method to type case the data based on modified data type. This method will used for
+   * retrieving the data after change in data type restructure operation
+   *
+   * @param data
+   * @param restructuredDataType
+   * @param currentDataOffset
+   * @param length
+   * @return
+   */
+  public static long getDataBasedOnRestructuredDataType(byte[] data, DataType restructuredDataType,
```

Comment:
This case won't exist from short to int or float to double?. Here I can see handling only for INT to LONG

Reply:
yes because alter datatype change is only for Int to BigInt and decimal from lower precision to higher precision. So float to double case is not required to be handled